scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/SynchronizationInternal.java

```diff
@@ -0,0 +1,145 @@
+
+    void clear(String originator, TaskId taskid, String channel) throws InvalidChannelException, IOException;
+
+    Set<String> keySet(String originator, TaskId taskid, String channel) throws InvalidChannelException;
+
+    <T extends Serializable> Collection<T> values(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException;
+
+    <T extends Serializable> Set<Map.Entry<String, T>> entrySet(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException;
+
+    <T extends Serializable> T getOrDefault(String originator, TaskId taskid, String channel, String key,
+            T defaultValue) throws InvalidChannelException;
+
+    void forEach(String originator, TaskId taskid, String channel, String action)
+            throws InvalidChannelException, CompilationException;
+
+    void replaceAll(String originator, TaskId taskid, String channel, String function)
+            throws InvalidChannelException, CompilationException, IOException;
+
+    <T extends Serializable> T putIfAbsent(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException;
+
+    <T extends Serializable> boolean remove(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException;
+
+    <T extends Serializable> boolean replace(String originator, TaskId taskid, String channel, String key, T oldValue,
+            T newValue) throws InvalidChannelException, IOException;
+
+    <T extends Serializable> T replace(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException;
+
+    <T extends Serializable> T computeIfAbsent(String originator, TaskId taskid, String channel, String key,
+            String mappingFunction) throws InvalidChannelException, CompilationException, IOException;
+
+    <T extends Serializable> T computeIfPresent(String originator, TaskId taskid, String channel, String key,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException;
+
+    <T extends Serializable> T compute(String originator, TaskId taskid, String channel, String key,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException;
+
+    <T extends Serializable> T merge(String originator, TaskId taskid, String channel, String key, T value,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException;
+
+    // Additional operations
+
+    <T extends Serializable> PredicateActionResult<T> conditionalCompute(String originator, TaskId taskid,
+            String channel, String key, String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)